src/platform/Linux/CHIPBluezHelper.h

```diff
@@ -205,6 +205,58 @@ CHIP_ERROR StopDiscovery(BluezEndpoint * apEndpoint);
 
 CHIP_ERROR ConnectDevice(BluezDevice1 * apDevice);
 
+/// Iterates over available BlueZ adapters
+///
+/// Usage example:
+///
+///  AdapterIterator iterator;
+///  while (iteerator.Next()) {
+///      std::cout << iterator.GetAddress() << std::endl;
+///  }
+class AdapterIterator
+{
+public:
+    ~AdapterIterator();
+
+    /// Moves to the next DBUS interface.
+    ///
+    /// MUST be called at least once
+    bool Next();
```

Comment:
This needs to document the return value's semantics, because right now those semantics are pretty weird, and I don't know whether that weirdness is intended or a bug.

Specifically, say `mObjectList` is a length-1 list and the one item is a bluez adapter.  The first call to `AdapterIterator::Next` will:
1. Initialize the iterator.
2. Fill in `mCurrent`
3. Update `mCurrentListItem` to be null.
4. Return `false`.

I _think_ that what should actually happen is that `Advance` should return true when `foundAdapter` ends up true and that `Next` should just `return Advance()`, but maybe the above behavior is purposeful for some reason?

Reply:
Changed Advance to return true/false.